src/cargo/ops/cargo_new.rs

```diff
@@ -962,11 +962,17 @@ fn update_manifest_with_new_member(
         )
     })?;
 
-    let display_path = relpath
-        .iter()
-        .filter_map(|comp| comp.to_str())
-        .collect::<Vec<_>>()
-        .join("/");
+    let mut components = Vec::new();
+    for comp in relpath.iter() {
+        let comp = comp.to_str().with_context(|| {
```

Comment:
nit: you can do this as `let components = relpath.iter().map(...).collect::<Result<_, _>>()?;` but its sometimes I pain to do it that way, so not required.